platform/drm/cog-platform-drm.c

```diff
@@ -352,6 +352,21 @@ check_drm(void)
     return supported;
 }
 
+static int32_t
+find_crtc_for_encoder(const drmModeRes *resources, const drmModeEncoder *encoder)
+{
+    for (int i = 0; i < resources->count_crtcs; i++) {
+        const uint32_t crtc_mask = 1 << i;
+        const uint32_t crtc_id = resources->crtcs[i];
+        if (encoder->possible_crtcs & crtc_mask) {
+            return crtc_id;
+        }
+    }
+
+    /* no match found */
+    return -1;
```

Comment:
This should return zero, because later below the code checks for `crtc_id != 0`.

Reply:
Fixed it by handing result correctly in calling function